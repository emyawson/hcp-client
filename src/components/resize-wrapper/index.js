export * from './resize-wrapper.component';
