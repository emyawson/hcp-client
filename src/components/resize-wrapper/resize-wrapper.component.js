import React from 'react';
import { Observable } from 'rxjs';
import styled from 'styled-components';

export const Wrapper = styled.div`
  flex: 1;
  display: flex;
  flex-basis: auto;
  flex-direction: column;
  min-height: inherit;
  justify-content: center;
`;
Wrapper.displayName = 'Wrapper';

export class ResizeWrapper extends React.Component {
  constructor(props) {
    super(props);
    const { minHeight = 0 } = props;
    this.state = {
      height: minHeight,
    };
  }

  componentDidMount() {
    this.setHeight();
    Observable.fromEvent(window, 'resize').subscribe(event => {
      this.setHeight();
    });
  }

  render() {
    const { render } = this.props;
    const { height } = this.state;
    return (
      <Wrapper innerRef={node => (this.wrapper = node)}>
        {render(height)}
      </Wrapper>
    );
  }

  setHeight = () => {
    const { wrapper } = this;
    const { disableResize = false } = this.props;
    if (wrapper && !disableResize) {
      const { resizeFunction } = this.props;
      this.setState({
        height: resizeFunction(wrapper.clientHeight),
      });
    }
  };
}
