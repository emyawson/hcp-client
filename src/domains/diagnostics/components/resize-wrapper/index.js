export { ResizeWrapper } from './resize-wrapper.component';
