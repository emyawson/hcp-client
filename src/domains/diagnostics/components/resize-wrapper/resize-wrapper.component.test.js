import React from 'react';
import { shallow, mount } from 'enzyme';

import { ResizeWrapper } from './resize-wrapper.component';

describe('Resize wrapper component', () => {
  const baseProps = {
    render: height => <div>sample</div>,
    resizeFunction: () => 3,
  };

  it('Renders correctly', () => {
    const mockProps = {
      ...baseProps,
      disableResize: false,
      minHeight: 0,
    };
    const wrapper = shallow(<ResizeWrapper {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Assigns wrapper correctly', () => {
    const root = mount(<ResizeWrapper {...baseProps} />);
    root.instance().setHeight();
    expect(root.instance().wrapper).toBe(root.find('Wrapper').getDOMNode());
  });

  it('Sets height correctly', () => {
    const mockProps = {
      ...baseProps,
      resizeFunction: height => height,
    };
    const root = mount(<ResizeWrapper {...mockProps} />);
    root.instance().setHeight();
    expect(root.state('height')).toEqual(root.instance().wrapper.clientHeight);
  });
});
