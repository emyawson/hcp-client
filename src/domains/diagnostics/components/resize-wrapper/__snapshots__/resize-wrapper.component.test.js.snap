// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resize wrapper component Renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResizeWrapper
    disableResize={false}
    minHeight={0}
    render={[Function]}
    resizeFunction={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        sample
      </div>,
      "innerRef": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": "sample",
      },
      "ref": null,
      "rendered": "sample",
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          sample
        </div>,
        "innerRef": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "sample",
        },
        "ref": null,
        "rendered": "sample",
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
